Lib/MigrationVersion.php

```diff
@@ -337,12 +348,25 @@ public function run($options) {
 			krsort($mapping);
 		}
 
+
 		foreach ($mapping as $version => $info) {
 			if (($direction === 'up' && $version > $targetVersion)
 				|| ($direction === 'down' && $version < $targetVersion)) {
 				break;
-			} elseif (($direction === 'up' && $info['migrated'] === null)
+			} elseif (($direction === 'up' && $info['migrated'] === null) 
 				|| ($direction === 'down' && $info['migrated'] !== null)) {
+				
+				$jumpVersion = $this->getVersionByName($mapping);
+				if ($version < $jumpVersion) {
+					$this->jump($version, $info['type']);
```

Comment:
``$info['type']`` gets re-used a few time. May be put it into a variable ``$type``?